File: nodejs_backend_with_db/commands.js
#!/usr/bin/env node

// Import required modules
const { Command } = require('commander');
const { prompt } = require('inquirer');

const { removeClassroom, listClassrooms } = require("./controller/Classroom")
const { addStudent, removeStudent, listStudents } = require("./controller/Student")
const { addClassroom } = require("./index");
const { scheduleAssignment, submitAssignment } = require('./controller/Assignment');

// Create a new Commander command
const program = new Command();

const questionsForClassroom = [
  {
    type: 'input',
    name: 'ClassName',
    message: 'Enter Name of classroom'
  }
];

const questionsForStudent = [
  {
    type: 'input',
    name: 'studentID',
    message: 'Enter the studentID'
  },
  {
    type: 'input',
    name: 'studentName',
    message: 'Enter the studentName'
  },
  {
    type: 'input',
    name: 'className',
    message: 'Enter the Name of classroom'
  },
]

const questionsForStudentRemove = [
  {
    type: 'input',
    name: 'StudentID',
    message: 'Enter the StudentID'
  },
  {
    type: 'input',
    name: 'className', 
    message: 'Enter the Name of classroom'
  }
];

const questionsForAssignment = [
  {
    type: 'input',
    name: 'AssignmentID',
    message: 'Enter the Assignment ID',
  },
  {
    type: 'input',
    name: 'AssignmentDetail',
    message: 'Enter the Assignment Detail',
  },
  {
    type: 'input',
    name: 'className',
    message: 'Enter the Classroom Name',
  },
];
const questionsForAssignmentSubmission = [
  {
    type: 'input',
    name: 'studentID',
    message: 'Enter the Student ID:',
  },
  {
    type: 'input',
    name: 'className',
    message: 'Enter the Class Name:',
  },
  {
    type: 'input',
    name: 'assignmentID',
    message: 'Enter the Assignment ID:',
  },
];

program
  .command('add_student')
  .alias('a')
  .description('Add a student')
  .action(() => {
    prompt(questionsForStudent).then(answers => {

      addStudent(answers.studentID, answers.studentName, answers.className);
    });
  });

program
  .command('add_classroom')
  .alias('a')
  .description('Add a classroom')
  .action(() => {
    prompt(questionsForClassroom).then(answers => addClassroom(answers.ClassName));
  });

program
  .command('remove_classroom')
  .alias('r')
  .description('Remove a classroom')
  .action(() => {
    prompt(questionsForClassroom).then(answers => removeClassroom(answers.ClassName));
  });

program
  .command('list_classrooms')
  .alias('l')
  .description('List all classrooms')
  .action(() => {
    listClassrooms();
  });

program
  .command('remove_student')
  .alias('rs')
  .description('Remove a student from a classroom')
  .action(() => {
    prompt(questionsForStudentRemove).then(answers => removeStudent(answers.StudentID, answers.className));
  });

program
  .command('list_students')
  .alias('ls')
  .description('List students in a classroom')
  .action(() => {
    prompt(questionsForClassroom).then(answers => listStudents(answers.ClassName));
  });
program
  .command('schedule_assignment')
  .alias('sa')
  .description('Schedule a new assignment')
  .action(() => {
    prompt(questionsForAssignment).then((answers) => {
      scheduleAssignment(answers.AssignmentID, answers.AssignmentDetail, answers.className);
    });
  });

program
  .command('submit_assignment')
  .alias('s')
  .description('Submit an assignment')
  .action(() => {
    prompt(questionsForAssignmentSubmission).then(answers => {
      // Call the submitAssignment function with the answers
      submitAssignment(answers.studentID, answers.className, answers.assignmentID);
    });
  });


program.parse(process.argv);

File: cpp_oops_without_db/Commands/commands.cpp
#include "../Header/VirtualClass.h"
#include<bits/stdc++.h>

using namespace std;

void processCommands(VirtualClass& vc)
{

    string className;
    string command;
    string assignment_detail;
    int student_id;
    int assignment_id;

    cout << endl;
    cout << "Here you can write commands like add_classroom, add_student, etc..." << endl;
    cout << "For details of all commands, you can type 'help' in the command line." << endl;
    cout << endl;

    while (true)
    {
        cout<<endl;
        cout << "Enter a command: ";
        cin >> command;

        if (command == "add_classroom")
        {
            cin >> className;
            vc.add_classroom(className);
        }
        else if (command == "add_student")
        {
            cin >> student_id;
            cin >> className;
            string student_name;
            cin.ignore(); 
            getline(cin, student_name);

           
            if (student_name.empty())
            {
                vc.add_student(student_id, className);
            }
            else
            {
                vc.add_student(student_id, className, student_name);
            }
        }
        else if (command == "list_students")
        {
            cin >> className;
            vc.list_students(className);
        }
        else if (command == "list_classrooms")
        {
            vc.list_classrooms();
        }
        else if (command == "remove_classroom")
        {
            cin >> className;
            vc.remove_classroom(className);
        }
        else if (command == "remove_student")
        {
            cin >> student_id;
            cin >> className;
            vc.remove_student(student_id, className);
        }
        else if (command == "schedule_assignment")
        {
            cin >> className;
            cin >> assignment_detail;
            vc.schedule_assignment(className, assignment_detail);
        }
        else if (command == "submit_assignment")
        {
            cin >> student_id;
            cin >> className;
            cin >> assignment_id;
            vc.submit_assignment(student_id, className, assignment_id);
        }
        else if (command == "help")
        {
            cout << "Available commands are:" << endl;
            cout << "add_classroom [class_name]" << endl;
            cout << "add_student [student_id] [class_name] [student_name] " << endl;
            cout << "list_students [class_name]" << endl;
            cout << "list_classrooms" << endl;
            cout << "remove_classroom [class_name]" << endl;
            cout << "remove_student [student_id] [class_name]" << endl;
            cout << "schedule_assignment [class_name] [assignment_detail]" << endl;
            cout << "submit_assignment [student_id] [class_name] [assignment_id]" << endl;
        }
        else if (command == "quit")
        {
            break; 
        }
        else
        {
            cout << "Invalid command. For available commands, type 'help'" << endl;
        }
    }

}
